varnish.m4

```diff
@@ -145,15 +145,17 @@ AC_DEFUN([_VARNISH_VMOD_LDFLAGS], [
 # _VARNISH_VMOD_CONFIG
 # --------------------
 AC_DEFUN([_VARNISH_VMOD_CONFIG], [
+	dnl Check the VMOD toolchain
+	AC_USE_SYSTEM_EXTENSIONS
```

Comment:
@Dridi could I please ask for your advice / opinion about this aspect?
To avoid issues with API headers accidentally making use of GNU/POSIX extensions, I would think that we should also, by default, enable them for vmods.
Yet with a sane vmod `configure.ac` like the one from [a certain development kip](https://github.com/Dridi/vcdk), this will lead to the following warnings:
```
configure.ac:22: warning: AC_COMPILE_IFELSE was called before AC_USE_SYSTEM_EXTENSIONS
aclocal.m4:9533: _VARNISH_VMOD_CONFIG is expanded from...
aclocal.m4:9579: _VARNISH_VMOD is expanded from...
aclocal.m4:9763: VARNISH_VMODS is expanded from...
configure.ac:22: the top level
configure.ac:22: warning: AC_LINK_IFELSE was called before AC_USE_SYSTEM_EXTENSIONS
aclocal.m4:9533: _VARNISH_VMOD_CONFIG is expanded from...
aclocal.m4:9579: _VARNISH_VMOD is expanded from...
aclocal.m4:9763: VARNISH_VMODS is expanded from...
configure.ac:22: the top level
configure.ac:22: warning: AC_CHECK_INCLUDES_DEFAULT was called before AC_USE_SYSTEM_EXTENSIONS
aclocal.m4:9533: _VARNISH_VMOD_CONFIG is expanded from...
aclocal.m4:9579: _VARNISH_VMOD is expanded from...
aclocal.m4:9763: VARNISH_VMODS is expanded from...
configure.ac:22: the top level
```
In turn, those can be avoided by the following change to call the `VARNISH_*` macros before any compiler/linker related macros

```diff
diff --git a/configure.ac b/configure.ac
index 178f2e9..2d9e75c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6,10 +6,6 @@ AC_CONFIG_HEADERS([config.h])
 
 AM_INIT_AUTOMAKE([1.12 -Wall -Werror foreign parallel-tests])
 AM_SILENT_RULES([yes])
-AM_PROG_AR
-
-LT_PREREQ([2.2.6])
-LT_INIT([dlopen disable-static])
 
 AC_ARG_WITH([rst2man],
        AS_HELP_STRING(
@@ -21,6 +17,11 @@ AC_ARG_WITH([rst2man],
 VARNISH_PREREQ([6.0.0])
 VARNISH_VMODS([test])
 
+AM_PROG_AR
+
+LT_PREREQ([2.2.6])
+LT_INIT([dlopen disable-static])
+
 AC_CONFIG_FILES([
        Makefile
        src/Makefile
```

but I am not sure if this is the best option.